README.md

```diff
@@ -840,6 +841,95 @@ public class Handler : IRequiresAsyncInitialization
 public class Container : IContainer<Server> {}
 ```
 
+### `Owned<T>` support
+
+Using `Owned<T>` instead of `T` gives a component the control over when the `T` instance and all its dependencies are released back to the container while maintaining the component’s agnosticism about all the relative lifespans. Injecting `Owned<T>` or delegates returning `Owned<T>` is enabled out of the box, and so are the `AsyncOwned<T>` versions of these. `AsyncOwned<T>` will be required to be used instead of `Owned<T>` when the `T` in question or any of its dependencies require asynchronous disposal.
+
+If you obtain an `Owned<T>`, you access the `T` instance using the `Value` property, and you dispose the `Owned<T>` instance to release the `T` instance back to the container. The container will never dispose the contained `T` instance automatically, not even when the container itself is disposed. It’s now your responsibility to call `Owned<T>.Dispose()` when you’re ready. If you don’t, the `T` instance and all its dependencies will never be cleaned up.
+
+The typical scenario is when you combine this with delegates by injecting `Func<Owned<Y>>` or `Func<X, Owned<Y>>` into a component. This enables the component to acquire and release `Y` instances as many times as it needs, possibly with varying parameters, and possibly overlapping with each other before releasing them back to the container. Here’s one such usage:
+
+```cs
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using StrongInject;
+using System;
+using System.Threading.Tasks;
+
+[Register(typeof(SomeViewModel))]
+[Register(typeof(SomeDbContext))]
+public partial class Container : IContainer<SomeViewModel>
+{
+    [Factory]
+    public static SqlConnection CreateSqlConnection()
+    {
+        throw new NotImplementedException();
+    }
+}
+
+public class SomeViewModel
+{
+    private readonly Func<Owned<SomeDbContext>> dbContextFactory;
+
+    public SomeViewModel(Func<Owned<SomeDbContext>> dbContextFactory)
+    {
+        this.dbContextFactory = dbContextFactory;
+    }
+
+    public async Task SaveAsync(string newName)
+    {
+        // Using Func<Owned<SomeDbContext>> instead of Func<SomeDbContext> ensures that the context will not be
+        // disposed while this method is still using it, even if SomeViewModel is released back to the container.
+
+        // Using Func<Owned<SomeDbContext>> instead of Owned<SomeDbContext> gives the ability for SaveAsync to
+        // get a new SomeDbContext instance that is not being used anywhere else, so overlapping calls to SaveAsync
+        // are no problem.
+
+        // By modeling the dependency this way, SomeViewModel remains agnostic about whether SomeDbContext is
+        // outliving it by being pooled and recycled (not shown here). Both SomeViewModel and SomeDbContext are
+        // even agnostic about whether the SqlConnection is being pooled and recycled.
```

Comment:
Actually, is pooling even possible with StrongInject? Maybe I should remove mentions of it if not.

Reply:
You could do this by registering SqlConnection via an `IFactory<T>`?

```csharp
PoolingSqlConnectionFactory : IFactory<SqlConnection>
{
    public SqlConnection Create() => Pool.Get();
    public void Release(SqlConnection ) => Pool.Return();
}
```

Reply:
Ah, nice!